### ISSUE TITLE
update "creating a napari plugin"

### ISSUE THREAD

**Issue by neuromusic** [napari/napari#2951]
## 📚 Documentation

With the imminent launch of [the napari hub](https://napari-hub.org) it looks like its time to update the "[creating a napari plugin](https://napari.org/plugins/stable/for_plugin_developers.html)" guide.

Recommended changes:

- [ ] discuss cookiecutter as "step 0" rather than at the end
- [ ] update references to the github tag or PyPI search to point to [the napari hub](https://napari-hub.org) 
- [ ] give plugin devs some guidance on [metadata to include](https://github.com/chanzuckerberg/napari-hub/blob/main/docs/customizing-plugin-listing.md) and [writing a napari-specific description](https://github.com/chanzuckerberg/napari-hub/blob/main/docs/writing-the-perfect-description.md)
- [ ] Using the napari hub CLI tool, [napari-hub-cli](https://github.com/chanzuckerberg/napari-hub-cli), to preview/check metadata before sharing

**Comment by liaprins-czi** [comment]
There are other changes to make to the napari.org site in addition to updating the "creating a napari plugin" guide page, outlined in this doc:
https://docs.google.com/document/d/1-O1nUl75jPxnH1v6xVewOetFpGPchtThVjI2cEWsvSs/edit#heading=h.ky26cnmlji7x

@neuromusic Shoudl these be a separate issue(s)?

- [ ] add link to hub within main menu
- [ ] [add a new “Finding and installing a napari plugin” page within the main “Plugins” section](https://docs.google.com/document/d/1-O1nUl75jPxnH1v6xVewOetFpGPchtThVjI2cEWsvSs/edit#heading=h.sw95mln84rk1)
- [ ] [update "Plugins" section copy](https://docs.google.com/document/d/1-O1nUl75jPxnH1v6xVewOetFpGPchtThVjI2cEWsvSs/edit#heading=h.2yjjm8qp5kly)
- [ ] replace link in footer to hub (not until new visual designs are applied)
- [ ] add sentence about napari hub at bottom of home page, along with the other sentences for the main menu sections (not until new visual designs are applied)
